app/javascript/vue/protocol/protocolMetadata.vue

```diff
@@ -17,7 +17,7 @@
           <span class="fas fa-print" aria-hidden="true"></span>
         </a>
         <button class="btn btn-light" @click="openVersionsModal">{{ i18n.t("protocols.header.versions") }}</button>
-        <button v-if="!protocol.attributes.published" @click="$emit('publish')" class="btn btn-primary">{{ i18n.t("protocols.header.publish") }}</button>
+        <button v-if="isPublishButtonVisible(protocol)" @click="$emit('publish')" class="btn btn-primary">{{ i18n.t("protocols.header.publish") }}</button>
```

Comment:
You can just check if `publish_url` exist. This value already include permission check.

```
def publish_url
    return unless can_publish_protocol_in_repository?(object)
    publish_protocol_path(object)
  end
```